src/python/WMCore/WMSpec/Steps/Executor.py

```diff
@@ -163,3 +158,26 @@ def post(self, emulator = None):
 
         """
         return None
+
+
+    def setCondorChirpAttrDelayed(self, key, value):
+        """
+        _setCondorChirpAttrDelayed_
+
+        Util to call condor_chirp and publish the key/value pair
+
+        """
+        # construct condor_chirp binary location from CONDOR_CONFIG
+        condor_chirp_bin = None
+        condor_config = os.getenv('CONDOR_CONFIG', None)
+        if condor_config:
+            condor_config_dir = os.path.dirname(condor_config)
+            condor_chirp_bin = os.path.join(condor_config_dir, 'main/condor/libexec/condor_chirp')
+            if not os.path.isfile(condor_chirp_bin):
+                condor_chirp_bin = None
+
+        if condor_chirp_bin:
+            args = [ condor_chirp_bin, 'set_job_attr_delayed', str(key), str(value.__repr__()) ]
```

Comment:
Is repr(value) guaranteed to return a string ? Each element of the argument list *has* to be a string, otherwise the subprocess.call() method will fail. Will change to repr(value) though.